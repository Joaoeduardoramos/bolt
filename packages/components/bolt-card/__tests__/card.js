const { render } = require('@bolt/twig-renderer');
const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(join(__dirname, '../card.schema.yml'));
const { tag } = schema.properties;

describe('<bolt-image>', async () => {
  test('<bolt-card>', async () => {
    const results = await render('@bolt-components-card/card.twig', {
      media: {
        image: {
          src: '/images/placeholders/landscape-16x9-mountains.jpg',
          alt: 'Image alt.',
        },
      },
      body: {
        eyebrow: 'This is an eyebrow',
        headline: 'This is a headline',
        paragraph: 'This is a paragraph.',
      },
      actions: [
        {
          text: 'This is a button',
          url: 'https://pega.com',
        },
      ],
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
