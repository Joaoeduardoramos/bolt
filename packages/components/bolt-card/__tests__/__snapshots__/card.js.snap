// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card basic card 1`] = `
"<bolt-card
     tag=\\"article\\"      theme=\\"none\\"      height=\\"full\\"         
>
    
    
  <replace-with-children  class=\\"c-bolt-card\\">
    
                          

<bolt-card-body>
  <replace-with-children class=\\"c-bolt-card__body\\">
                      
                            


  

 
  




  

  





<p  class=\\"c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--medium\\">
  
      This is a paragraph.
  
  </p>
            </replace-with-children>
</bolt-card-body>
                  </replace-with-children>
</bolt-card>"
`;
